### PR TITLE
New directory structure with git submodule

### PR DESCRIPTION
**Pull request by EmmanuelObua**

### CODE REVIEW

File: vue-element-admin/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#! /bin/env bash
+
+set -e # exit when any command fails
+
+PUBLIC_URL=/vue-element-admin/vue-element-admin npm install && npm run build:prod
```
Comment (fregante) [inline vue-element-admin/build.sh:5]: I think this:

```suggestion
cd vue-element-admin
npm install
PUBLIC_URL=/vue-element-admin npm run build:prod
```

File: vue-element-admin/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#! /bin/env bash
+
+set -e # exit when any command fails
+
+cd vue-element-admin
+npm install
+PUBLIC_URL=/vue-element-admin npm run build:prod
```
Comment (fregante) [inline vue-element-admin/build.sh:7]: Great! The deployment worked. Now the problem is that `PUBLIC_URL` is not recognized by their build, so we should find a way to customize that URL — if it exists

Comment (EmmanuelObua) [reply vue-element-admin/build.sh:10]: Yes, still trying out something by modify the vue.config.js

Comment (EmmanuelObua) [reply vue-element-admin/build.sh:10]: It's now working though you have to append index.html at the end just like in the link below 
[Vue-element-admin](https://playground-b1751et14-pixiebrix.vercel.app/vue-element-admin/index.html)
index file not being read automatically as it should be

Comment (twschiller) [reply vue-element-admin/build.sh:10]: Hi @EmmanuelObua, I think you have to add a rewrite rule to vercel.json like we did for the react project: http://github.com/pixiebrix/playground/blob/3c5495d846f1d2852e82ac04a090c22ff312acc7/vercel.json#L5-L5

**Comment by fregante** [comment]
I think it's working now! 

https://playground-d4d8lvajw-pixiebrix.vercel.app/vue-element-admin/#/icon/index

The changes in `vercel.json` should not be necessary because it uses `#/`.

The only issue I see is that it doesn't work when opening it from root: https://playground-d4d8lvajw-pixiebrix.vercel.app

This is because the URL is `/vue-element-admin` but it must be `/vue-element-admin/`. You can actually use vercel.json to do this redirect, like this: https://vercel.com/support/articles/does-vercel-support-permanent-redirects#other-redirects

**Comment by twschiller** [comment]
Hmm, I tried implementing the redirect. What's interesting is that without the slash it's still getting the html file, it's just that the links to the static resources are wrong...

I'm OK to get this merged in an then we can try to tweak the rules

image

Something that was surprising to me was the source `"/vue-element-admin/vue-element-admin/(.*)"`. Why does that have 2 levels in there?

**Comment by fregante** [comment]
> What's interesting is that without the slash it's still getting the html file

That's because:

1. this PR [set the public path to `''`](https://github.com/pixiebrix/playground/blob/1f2f7da1dc0621635ce721d34ae6964c4ab78eeb/vue-element-admin/vue.config.js#L27)
2. this generated relative urls like `'static/file.js'`
3. on `/vue-admin` that relative url resolves to `/static/file.js` ❌ 
3. on `/vue-admin/` that relative url resolves to `/vue-admin/static/file.js` ✅ 

This was fixed by setting the public path to `'/vue-admin-element/' so urls would be _relative to root_ instead of _relative to the current path_

See https://github.com/pixiebrix/playground/commit/6d4734bd73e2d6ae4b3303fc9a0df3a40e71f474



> Something that was surprising to me was the source `"/vue-element-admin/vue-element-admin/(.*)"`.

That public URL doesn't exist so it never matched.

**Comment by twschiller** [comment]
Thanks @fregante. This URL `https://pbx.vercel.app/vue-element-admin` actually works for me now. Maybe there's a slight difference in behavior between the production vs. PR deployments? Or maybe something weird is going on with my browser caching?

I created https://github.com/pixiebrix/playground/issues/27 to remove the unused vercel.json routing entry

**Pull request merged**